servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/H2ResponseCancelTest.java

```diff
@@ -0,0 +1,188 @@
+    private final CountDownLatch secondRequestReceivedLatch = new CountDownLatch(1);
+
+    private final CountDownLatch firstResponseLatch = new CountDownLatch(1);
+    private final CountDownLatch secondResponseLatch = new CountDownLatch(1);
+
+    private final AtomicInteger newConnectionsCounter = new AtomicInteger();
+
+    public H2ResponseCancelTest() {
+        super(CACHED, CACHED_SERVER);
+        protocol(HTTP_2.config);
+        serviceFilterFactory(service -> new StreamingHttpServiceFilter(service) {
+            @Override
+            public Single<StreamingHttpResponse> handle(HttpServiceContext ctx, StreamingHttpRequest request,
+                                                        StreamingHttpResponseFactory responseFactory) {
+                try {
+                    if ("first".equals(request.queryParameter(PARAM))) {
+                        firstRequestReceivedLatch.countDown();
+                        firstResponseLatch.await();
+                    }
+                    if ("second".equals(request.queryParameter(PARAM))) {
+                        secondRequestReceivedLatch.countDown();
+                        secondResponseLatch.await();
+                    }
+                    if ("close".equals(request.queryParameter(PARAM))) {
+                        ctx.closeAsync().toFuture().get();
+                    }
+                } catch (Exception e) {
+                    return failed(e);
+                }
+                return delegate().handle(ctx, request, responseFactory);
+            }
+        });
+        connectionFactoryFilter(factory -> new DelegatingConnectionFactory<InetSocketAddress,
+                FilterableStreamingHttpConnection>(factory) {
+            @Override
+            public Single<FilterableStreamingHttpConnection> newConnection(InetSocketAddress inetSocketAddress,
+                                                                           @Nullable TransportObserver observer) {
+                return defer(() -> {
+                    newConnectionsCounter.incrementAndGet();
+                    return delegate().newConnection(inetSocketAddress, observer);
+                });
+            }
+        });
+    }
+
+    @Test
+    public void testClient() throws Exception {
+        // Release existing connection to avoid creating unexpected number of new h2 connections because of randomness
+        // in RRLB (it may randomly pick the reserved connection until all attempts are taken).
+        ((ReservedStreamingHttpConnection) streamingHttpConnection()).releaseAsync().toFuture().get();
+
+        int connectionsBefore = newConnectionsCounter.get();
+        requestCancellationResetsStreamButNotParentConnection(streamingHttpClient());
+        int connectionsAfter = newConnectionsCounter.get();
+        assertThat("Client unexpectedly created more connections instead of reusing the existing one",
+                connectionsAfter - connectionsBefore, is(0));
```

Comment:
nit: `assertThat(connectionsAfter, is(connectionsBefore));` or `assertEquals(connectionsBefore, connectionsAfter)` will give a more informative error message if a failure occurs.